src/Raven.Studio/typescript/viewmodels/database/settings/databaseSettings.ts

```diff
@@ -0,0 +1,526 @@
+        summaryGrid.headerVisible(true);
+
+        summaryGrid.init(() => this.fetchSummaryData(), () => {
+            if (this.hasPendingValues()) {
+                return [
+                    new textColumn<models.settingsEntry>(summaryGrid, x => x.keyName(), "Configuration Key", "30%", {
+                        sortable: "string",
+                        extraClass: x => x.entryClassForSummaryMode()
+                    }),
+                    new textColumn<models.settingsEntry>(summaryGrid, x => x.effectiveValueInUseText(), "Effective Value in Use", "30%", {
+                        sortable: "string",
+                        extraClass: (x) => x.effectiveValueInUseText() ? "value-has-content" : ""
+                    }),
+                    new textColumn<models.settingsEntry>(summaryGrid, x => x.pendingValueText() , "Pending Value", "30%", {
+                        sortable: "string",
+                        extraClass: (x) => x.pendingValueText() ? "value-has-content" : ""
+                    })
+                ]
+            } else {
+                return [
+                    new textColumn<models.settingsEntry>(summaryGrid, x => x.keyName(), "Configuration Key", "30%", {
+                        sortable: "string",
+                        extraClass: x => x.entryClassForSummaryMode()
+                    }),
+                    new textColumn<models.settingsEntry>(summaryGrid, x => x.effectiveValue(), "Effective Value", "40%", {
+                        sortable: "string",
+                        extraClass: (x) => x.effectiveValue() ? "value-has-content" : ""
+                    }),
+                    new textColumn<models.settingsEntry>(summaryGrid, x => x.effectiveValueOrigin(), "Origin", "20%", {
+                        sortable: "string",
+                        extraClass: (x) => {
+                            let classes = "source-item ";
+                            switch (x.effectiveValueOrigin()) {
+                                case "Database":
+                                    classes += "source-item-database";
+                                    break;
+                                case "Server":
+                                    classes += "source-item-server";
+                                    break;
+                                case "Default":
+                                    classes += "source-item-default";
+                                    break;
+                            }
+                            return x.effectiveValue() ? classes : classes += " null-value";
+                        }
+                    })
+                ]
+            }
+        });
+    }
+    
+    private fetchSummaryData(): JQueryPromise<pagedResult<models.settingsEntry>> {
+        let entriesForSummaryMode = _.filter(this.allEntries(), x => x.showEntry());
+
+        entriesForSummaryMode = entriesForSummaryMode.reduce((acc,element) => {
```

Comment:
here you concat tables multiple times. 

I'd rather filter this table twice and concat one time. Also try to avoid using lodash, js array has filter method. 

I don't remeber if we can avoid logic based on css classes?

Reply:
done